plugins/mcp/plugin_test.go

```
@@ -0,0 +1,691 @@
+		if response == nil || len(response.Choices) == 0 {
+			t.Fatal("Expected response with choices")
+		}
+
+		// In non-agentic mode, the response should be a tool message
+		choice := response.Choices[0]
+		if choice.Message.Role != schemas.ModelChatMessageRoleTool {
+			t.Errorf("Expected tool message role, got: %s", choice.Message.Role)
+		}
+
+		client.Cleanup()
+		plugin.Cleanup()
```

golangci-lint (1.64.8)

Error return value of `plugin.Cleanup` is not checked

(errcheck)